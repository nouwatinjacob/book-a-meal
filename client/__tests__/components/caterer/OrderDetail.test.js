import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  OrderDetail,
  mapStateToProps
} from '../../../src/components/caterer/OrderDetail.jsx';
import {
  getAnOrderResponse
} from '../../__mockData__/mockOrder';

const props = {
  getAnOrderAction: jest.fn(),
  orderState: {
    order: getAnOrderResponse
  },
  match: {
    params: {
      orderId: 1
    }
  }
};

describe('OrderDetail component', () => {
  localStorage.setItem('token', catererToken);
  const setup = () => shallow(<OrderDetail {...props} />);

  it('should render OrderDetail component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });
});
