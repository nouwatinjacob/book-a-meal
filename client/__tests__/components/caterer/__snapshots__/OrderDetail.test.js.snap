// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetail component should render OrderDetail component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <div
      className="reduced-container"
    >
      <div
        className="row"
      >
        <div
          className="c-medium-12 c-small-12"
          id="pd-0"
        >
          <h6>
            September 6th 2018, 2:32:06 am
          </h6>
          <h5>
            Order No: 
            <strong>
              aa604a40-b174-11e8-bb0f-15bfecd2c56e
            </strong>
          </h5>
          <h5>
            Name: 
            <strong>
              Jacob Nouwatin
            </strong>
          </h5>
        </div>
      </div>
      <br />
      <table>
        <tbody>
          <tr>
            <th>
              Meal
            </th>
            <th>
              Quantity
            </th>
            <th>
              Unit Price
            </th>
            <th>
              Total
            </th>
          </tr>
          <tr>
            <td>
              Fried Rice and chicken Lap
            </td>
            <td>
              4
            </td>
            <td>
              3000
            </td>
            <td>
              12000
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
